### PR TITLE
Fixes the multi-gpu tests

### PR DESCRIPTION
**Pull request by robertgshaw2-neuralmagic**
SUMMARY:
- fixes kernel tests by skipping fp8 tests on multi-gpu for `test_cache.py` and `test_attention.py`
- fixes pos_encoding tests by running with `--forked`
- fixes sampler tests for multi-gpu case

TEST PLAN:
- runs in remote push automation

Note: opening a PR upstream to address sync issues

### CODE REVIEW

File: tests/samplers/test_rejection_sampler.py [hunk 1]
```diff
@@ -26,15 +26,15 @@ def mock_causal_accepted_tensor(
 
     accepted = (torch.arange(k).expand(batch_size, k) <=
                 last_accepted_indices.unsqueeze(-1).broadcast_to(
-                    batch_size, k)).to(device="cuda")
+                    batch_size, k)).to(device=device)
```
Comment (andy-neuma) [inline tests/samplers/test_rejection_sampler.py:29]: thanks!

**Review by andy-neuma — APPROVED** [review]
cool

**Review by tlrmchlsmth — APPROVED** [review]
lgtm when green

File: .github/scripts/run-tests [hunk 2]
```diff
@@ -58,7 +58,9 @@ do
 
     # this is a bit messy and brittle, but certain tests
     # need to be run with specific options
-    if [[ "${TEST}" == *"kernels"* ]]; then
+    if [[ "${TEST}" == *"kernels/test_pos_encoding"* ]]; then
```
Comment (andy-neuma) [inline .github/scripts/run-tests:61]: it looks like we need to restrict "tests/lora/test_layers.py" as well to run on just one device, i.e. `CUDA_VISIBLE_DEVICES=0 pytest tests-outside/lora/test_layers.py` ...

**Review by andy-neuma — APPROVED** [review]

**Review by mgoin — APPROVED** [review]

**Pull request merged**